src/middlewares/auth/authStrategies.js

```diff
@@ -29,12 +46,45 @@ const verifyGoogle = async (
   profile,
   done
 ) => {
-  const role = await UserRoles.findOne({ name: 'User' }).exec();
+  const user = await checkUser(profile);
+  const status = await getStatus();
+
+  if (user && user.status.toString() === status._id.toString()) {
+    return done(null, {
+      role: user.role,
+      name: user.name.concat(` ${user.surname}`),
+      id: user._id
+    });
+  }
 
-  profile.role = role._id.toString();
-  profile.name = profile.displayName;
+  const newUserData = {
+    body: {
+      name: profile.given_name,
+      middleName: 'Not provided',
+      surname: profile.family_name,
+      password: profile.id,
+      phoneNumber: 'Not provided',
+      address: 'Not provided',
+      email: profile.email
+    }
+  };
+
+  const res = null;
```

Comment:
https://stackoverflow.com/questions/31097312/how-to-pass-http-response-object-part-of-node-js-passport-module-based-authentic

Reply:
@VasylLymych, the way it is now works appropriately. Would it be better if I try to do it another way? What are the advantages of doing it another way (for example like the StackOverflow example)?

Reply:
Because this is the proper one. What if there is an error during DB call? How res.boom.badImplementation() method will be called if you pass res object as null?

Reply:
When you have time, if you don't mind, please explain how can I achieve this, because I don't understand how the appropriate way can be achieved. Thanks!

Reply:
So you have request object. You can do next:
req.body = {
      name: profile.given_name,
      middleName: 'Not provided',
      surname: profile.family_name,
      password: profile.id,
      phoneNumber: 'Not provided',
      address: 'Not provided',
      email: profile.email
    };
    
    const newUser = await usersService.createUser(req, req.res)

Reply:
```
const req = {
    body: {
      name: profile.given_name,
      middleName: 'Not provided',
      surname: profile.family_name,
      password: profile.id,
      phoneNumber: 'Not provided',
      address: 'Not provided',
      email: profile.email
    }
  };

  const newUser = await usersService.createUser(req, req.res);
 ```
 I get the same error: ```TypeError: Cannot read properties of undefined (reading 'json')```

Reply:
What is this: const req = {...}? You have the request object among verifyGoogle params...

Reply:
```
request.body = {
    name: profile.given_name,
    middleName: 'Not provided',
    surname: profile.family_name,
    password: profile.id,
    phoneNumber: 'Not provided',
    address: 'Not provided',
    email: profile.email
  };

  const newUser = await usersService.createUser(request.body, request.res);
```
This way the strategy fails and I get ```Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client```

Reply:
I submitted another commit (the latest one) where I deal with the problem the way I know.

Reply:
Actually this is happening because you are calling res.json method in service and after this we cannot change request object. The possible solution here can be to add next:
1) Add property to request req.isGoogleUser = true.
2) Add check to service: return newUser ? (req.isGoogleUser ? true : res.json(true)) : res.boom.badImplementation() here https://github.com/Sf-62-NodeJS/clothes-market-server/blob/66d7cb9009829601f009db0284b9b8a2c4292bd4/src/services/users.js#L177

Reply:
Done. Thanks!